tests/robot-tests/scripts/create_snapshots.py

```diff
@@ -118,6 +117,45 @@ def create_data_catalogue_snapshot(public_url) -> str:
 
     return json.dumps(result, sort_keys=True, indent=2)
 
+def create_all_methodologies_snapshot(public_url) -> str:
+    all_methodologies_url = f"{public_url.rstrip('/')}/methodology"
+    parsed_html = _gets_parsed_html_from_page(all_methodologies_url)
+
+    methodologies_accordion = parsed_html.find(id="publications")
+    if methodologies_accordion is None:
+        return []
+    
+    theme_sections = methodologies_accordion.select('[data-testid="accordionSection"]') or []
+
+    result = []
+
+    for theme_index, theme_html in enumerate(theme_sections):
+        theme = {
+            'theme_heading': theme_html.select_one(f'#publications-{theme_index + 1}-heading').string,
+            'topics': []
+        }
+        topics = theme_html.select('details') or []
+
+        for topic_html in topics:
+            topic = {
+                'topic_heading': topic_html.select_one('[id^="details-heading-"]').string,
+                'methodologies': []
+            }
+
+            methodologies = topic_html.select('[id^="methodology-heading-"]') or []
+            for methodology_html in methodologies:
+                methodologies = []
+                methodologies.append(methodology_html.string)
+                
+            topic['methodologies'].append(methodologies)
```

Comment:
So looking at this again, I think L147-150 can be removed and instead you can just have
```
            for methodology_html in methodologies:
                topic['methodologies'].append(methodology_html.string)
```

Reply:
Incidentally, if you're feeling really fancy, you could use list comprehension and get rid of this for loop completely:
https://www.w3schools.com/python/python_lists_comprehension.asp